CMake/deps/asan.cmake

```diff
@@ -0,0 +1,9 @@
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```

Comment:
Aren't the sanitizer options available in GCC?
[rippled also uses these](https://github.com/XRPLF/rippled/blob/fe05b8c4feb815e1aa6afd7ff8c85dbc92bcd651/Builds/CMake/RippledSettings.cmake#L62) (not saying it's necessarily a better way to include it.)

Reply:
Seems like you are right! i usually used `clang` and did not know `gcc` had them too. That's awesome.
Also, yes, the way `rippled` adds them is probably better than my quick and dirty way.

Would you take the time to add them properly instead of using what i did?